KeyError thrown from test_closed

the `test_closed` unit test is failing on Windows because the errno attribute is set to zero (the winerror is used instead) and the `_raise_ssl_error` function asks for the `errorcode[0]` which throws a KeyError.

```

======================================================================
FAIL: test_closed (OpenSSL.test.test_ssl.ConnectionSendallTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\konstantinos\documents\github\pyopenssl\OpenSSL\test\test_ssl.py", line 1937, in test_closed
    exc = self.assertRaises(SysCallError, server.sendall, b"hello, world")
  File "c:\users\konstantinos\documents\github\pyopenssl\OpenSSL\test\util.py", line 266, in failUnlessRaises
    exception.__name__,
AssertionError: <type 'exceptions.KeyError'> raised instead of SysCallError

----------------------------------------------------------------------
Ran 338 tests in 8.555s

FAILED (failures=1)
```

--- a/OpenSSL/SSL.py
+++ b/OpenSSL/SSL.py
@@ -1,4 +1,4 @@
-
+from sys import platform
 from functools import wraps, partial
 from itertools import count
 from weakref import WeakValueDictionary
@@ -856,8 +856,11 @@ class Connection(object):
         elif error == _lib.SSL_ERROR_SYSCALL:
             if _lib.ERR_peek_error() == 0:
                 if result < 0:
-                    raise SysCallError(
-                        _ffi.errno, errorcode[_ffi.errno])
+                    if platform == "win32":
+                        errno = _ffi.getwinerror()[0]
+                    else:
+                        errno = _ffi.errno
+                    raise SysCallError(errno, errorcode[errno])
                 else:
                     raise SysCallError(-1, "Unexpected EOF")
             else:

--- a/OpenSSL/test/test_ssl.py
+++ b/OpenSSL/test/test_ssl.py
@@ -6,7 +6,7 @@ Unit tests for :py:obj:`OpenSSL.SSL`.
 """
 
 from gc import collect, get_referrers
-from errno import ECONNREFUSED, EINPROGRESS, EWOULDBLOCK, EPIPE
+from errno import ECONNREFUSED, EINPROGRESS, EWOULDBLOCK, EPIPE, ESHUTDOWN
 from sys import platform, version_info
 from socket import SHUT_RDWR, error, socket
 from os import makedirs
@@ -1935,7 +1935,10 @@ class ConnectionSendallTests(TestCase, _LoopbackMixin):
         server, client = self._loopback()
         server.sock_shutdown(2)
         exc = self.assertRaises(SysCallError, server.sendall, b"hello, world")
-        self.assertEqual(exc.args[0], EPIPE)
+        if platform == "win32":
+            self.assertEqual(exc.args[0], ESHUTDOWN)
+        else:
+            self.assertEqual(exc.args[0], EPIPE)
 
 
 

